### PR TITLE
Remove deprecated functions

### LINKED ISSUE #182: New CodeSniffer errors
The last few PR reviews on Travis have reported a new batch of errors:

```
PHP CODE SNIFFER VIOLATION SOURCE SUMMARY
----------------------------------------------------------------------
SOURCE                                                           COUNT
----------------------------------------------------------------------
WordPress.PHP.DevelopmentFunctions.error_log                     2
WordPress.WP.DeprecatedFunctions.get_current_theme               1
WordPress.WP.DeprecatedFunctions.like_escape                     1
WordPress.WP.DeprecatedFunctions.get_themes                      1
----------------------------------------------------------------------
A TOTAL OF 5 SNIFF VIOLATIONS WERE FOUND IN 4 SOURCES
----------------------------------------------------------------------
```
Full report: https://travis-ci.org/MITLibraries/MITlibraries-parent/jobs/189373049

These either need to be escaped or fixed.

### PR DESCRIPTION
**Pull request by matt-bernhardt**
The recent upgrade of PHP's CodeSniffer has introduced a few new checks which are failing in our theme. For example, see [this recent Travis run](https://travis-ci.org/MITLibraries/MITlibraries-parent/jobs/190636799). Specifically:

```
PHP CODE SNIFFER VIOLATION SOURCE SUMMARY
----------------------------------------------------------------------
SOURCE                                                           COUNT
----------------------------------------------------------------------
Squiz.PHP.DisallowMultipleAssignments.Found                      34
WordPress.WP.I18n.MissingTranslatorsComment                      24
WordPress.PHP.DevelopmentFunctions.error_log                     2
WordPress.WP.DeprecatedFunctions.get_current_theme               1
WordPress.WP.DeprecatedFunctions.like_escape                     1
WordPress.WP.DeprecatedFunctions.get_themes                      1
----------------------------------------------------------------------
A TOTAL OF 63 SNIFF VIOLATIONS WERE FOUND IN 6 SOURCES
----------------------------------------------------------------------
```

To address the violations individually:

- Squiz.PHP.DisallowMultipleAssignments.Found

There's a number of these, but most of them boil down to something like https://github.com/MITLibraries/MITlibraries-parent/blob/9cdb0457ae07772851aa34edb459d178e5487830/page-hours-json.php#L184  
```
...
		  </span> </th>
		<?php $i = 2;
$day = $arDays[ $i ]; ?>
...
```
Basically you have to open the PHP block on one line, and then take the first action on the next:
```
...
		  </span> </th>
		<?php
$i = 2;
$day = $arDays[ $i ]; ?>
...
```

- WordPress.WP.I18n.MissingTranslatorsComment
This is clarified at https://codex.wordpress.org/I18n_for_WordPress_Developers#Descriptions - when using translation functions, it is preferred to add a comment immediately beforehand to give a hint to people building translations. I'm not sure how much attention our themes will get from translators, but these seem harmless enough.

- The rest of the violations are one-off changes that were solved in the first few commits - mostly be removing part of a rarely-if-ever used page.

This closes #182.

### CODE REVIEW

File: lib/hours.php [hunk 1]
```diff
@@ -1029,7 +1029,7 @@ function locationNameToIdOriginal( $name ) {
 function title_filter( $where, &$wp_query ) {
 	global $wpdb;
 	if ( $search_term = $wp_query->get( 'search_prod_title' ) ) {
-		$where .= ' AND ' . $wpdb->posts . '.post_title LIKE \'%' . esc_sql( like_escape( $search_term ) ) . '%\'';
+		$where .= ' AND ' . $wpdb->posts . '.post_title LIKE \'%' . esc_sql( $wpdb->esc_like( $search_term ) ) . '%\'';
```
Comment (matt-bernhardt) [inline lib/hours.php]: This change is most likely not needed, in that the file as a whole is unused (this was part of the old hours system that can probably just be removed)

**Comment by frrrances** [comment]
@matt-bernhardt thanks for the helpful description. Looks like you did some great cleanup! :shipit:
